### ISSUE TITLE
[BUG] Show on a specific display keeps changing.

### ISSUE THREAD

**Issue by ejgavin** [TheBoredTeam/boring.notch#174]
**Describe the bug**
In this bug, whenever my monitor is turned back on along with my MacBook, it switches to the monitor in the setting of show on a specific display.

**To Reproduce**
1. Choose built-in display as display to show Boring Notch on.
2. Turn on monitor.

**Expected behavior**
It should stay on the selected choice for the setting.

**Additional context**
On Glowing Panda (Snooty) 2.1 and MacOS Sequoia.

**Comment by CyanideSunryz** [comment]
Issue still occurs on 2.2.

**To reproduce:**
1. Connect the laptop to an external display (in my case, a projector)
2. BoringNotch moves to the external display
3. In settings, change it back to built-in retina display
4. Disconnect external display
5. Reconnect external display
6. BoringNotch has changed its display back to the external device

**Expected behaviour:**
It should retain the updated "Show on specific display" setting

**Additional context**
Running Macbook Air M2 15" with Sequoia 15.1.

**Comment by bronisMateusz** [comment]
The same thing happens when you put the computer to sleep mode with the display connected and wake it up. Then almost always the boring notch jumps to the external screen and does not remain on the built-in screen.

**Comment by CyanideSunryz** [comment]
Surely the default behaviour should be to display on the built-in screen since the external monitor is highly unlikely to have a real notch? I would imagine over 99% of installs are to use the dead space there rather than to have a functional blob at the top of an external monitor?

**Comment by mojst3r** [comment]
Issue also happens when using 2 external monitors, the notch jumps from one display to the other when waking up from sleep. The monitor that wakes up first gets the notch.

**Comment by bronisMateusz** [comment]
Yup, the problem still exists. Even after a clean reinstallation of the app.

**Comment by CyanideSunryz** [comment]
Still exists on 2.3

**Comment by LucasAschenbach** [comment]
I have the same issue. In my case this happens because when connected to a power supply, the macbook goes automatically into calmshell mode when closed which changes the main display and then overrides the `preferredScreen` variable. https://github.com/TheBoredTeam/boring.notch/blob/bed49dd90298d6335dc19326df4d7b5a19cf18e5/boringNotch/BoringViewCoordinator.swift#L74

**Temporary fix:** Either manually put your computer into sleep-mode using the menu or unplug your computer before closing it.

Ideal would be an additional option in the settings to either fix the preferred display or disable boring.notch for displays without a notch.

**Comment by kgmills** [comment]
This also happens with a KVM switch. When I switch from my Windows PC to MacBook BoringNotch always assigns one of the external as a new display. 

Note: This might also be because I need to change the input of one of my monitors from DP->HDMI due to MacOS issues.

**Comment by Alexander5015** [comment]
Should be fixed in the latest RC
